gpytorch/lazy/lazy_evaluated_kernel_tensor.py

```diff
@@ -24,6 +24,14 @@ def __init__(self, kernel, x1, x2, squeeze_row=False, squeeze_col=False, **param
         self.is_batch = self.x1.ndimension() == 3 or (self.x1.ndimension() == 2 and self.squeeze_row)
         self.params = params
 
+    @property
```

Comment:
I agree, let's make a separate issue out of this. One of the drawbacks of all the lazy evaluation we do is that very simple things like size or device mismatches often fail deep in the internals in several nested lazy tensors -- anything we can do to minimize this so that users don't have to understand how lazy tensors work to use GPs would be great.